transport/api/src/p2p.rs

```diff
@@ -558,14 +558,15 @@ pub async fn p2p_loop<T>(
                         METRIC_TRANSPORT_P2P_OPEN_CONNECTION_COUNT.increment(1.0);
                     }
 
-                    // TODO: async await in the event dispatch loop can block it, the await should be removed
                     if allowed_peers.contains(&peer_id) {
-                        // TODO: async spawn local and remove the RWLOCK
-                        if let Err(e) = network.add(&peer_id, PeerOrigin::IncomingConnection, vec![]).await {
-                            error!("transport - p2p - failed to update the record for '{peer_id}': {e}")
-                        }
+                        let network = network.clone();
+                        async_std::task::spawn(async move {
```

Comment:
How's the JoinHandle (now) discarded here going to be eventually awaited?

Reply:
Does it need to be? There were no execution follow ups in the original code.